test/functional/terminal/tui_spec.lua

```diff
@@ -197,26 +201,97 @@ describe('TUI', function()
     ]])
   end)
 
-  it('can handle arbitrarily long bursts of input', function()
-    -- Need extra time for this test, specially in ASAN.
-    screen.timeout = 60000
-    feed_command('set ruler')
+  it('paste: big burst of input', function()
```

Comment:
this test and the `paste: exactly 64 bytes` test have strange failures. The screen failure shows out-of-order lines, but I think this is a display/refresh issue. Logging shows that the lines are received by `nvim_put` in the correct order.

Reply:
Also locally for you?  (they work for me)

Reply:
Does the buffer contents look ok afterwards?

Reply:
> Also locally for you? (they work for me)

Sometimes. It's random.

> Does the buffer contents look ok afterwards?

Since it's a child Nvim can't use `helpers.expect()`. I will check by writing to a file.

Reply:
Doing `PUT_CURSEND` always seems to fix it. I guess the cursor placement depends on redraws which may be skipped/deferred? And this can probably be forced `check_cursor()`? But maybe it's fine if `nvim_put` always does `PUT_CURSEND`.

Reply:
Cursor position should never depend on when a redraw is done. Forcing PUT_CURSEND in normal mode breaks using `nvim_put` to replace `call setreg("x", data, mode) | normal "xp`, which I would hope it could do. Might better to have a flag for "always cursend" if this is needed for modelesslike TUI/GUI paste.